pkg/narrowcache/client.go

```diff
@@ -138,10 +126,54 @@ func (c *Client) updateCache(ctx context.Context, key string, watcher watch.Inte
 }
 
 func (c *Client) setToCache(key string, obj runtime.Object) error {
+	// cleanup unecessary fields
```

Comment:
could you move this part into the `GVKInfo` struct e.g. as a new `Cleanup` callback?
Because the `narrowcache` client is designed in a way to allow callers to extend what types are managed, by adding their own `GVKInfo` if needed, and this big switch kind of break the pattern.

Also, I think emptying `SetManagedFields` could be called regardless of the type, as it's in `ObjectMeta`.